server/src/main/java/org/opensearch/identity/ServiceAccount.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity;
+
+import java.security.Principal;
+import java.util.List;
+
+/**
+ * This interface defines a ServiceAccount.
+ *
+ * A ServiceAccount is a principal which can retrieve its associated permissions.
+ */
+public interface ServiceAccount extends Principal {
+
+    public List<String> getPermissions();
```

Comment:
Hey @scrawfor99, I wanted to jot down some thoughts I had recently about defining permissions and wanted to get your reaction.

I think the plugin developers should define the permissions that will make their plugins work, but on installation the cluster admin should be presented with the list of permissions the plugin/extension is requesting and consent to them to finish installation.

Similar to being presented the list of JSM permissions with `plugin-security.policy`, there could also be a companion file called `plugin-security.yml` to define security settings for the plugin.

For Example:

```
reserved_indices: [".opendistro-anomaly-results*", ".opendistro-anomaly-detector*", ".opendistro-anomaly-checkpoints", ".opendistro-anomaly-detection-state"]
permissions:
    cluster_permissions:
      - "cluster_monitor"
      - "cluster_composite_ops_ro"
      index_permissions:
      - index_patterns:
        - "*"
        allowed_actions:
        - "read"
```

The above policy would allow the plugin full access to those reserved indices and for any other action or actions on indices other than the reserved indices it would use those permissions defined.

I have been able to create and evaluate privileges with ephemeral roles in the security plugin and its possible not to have to create a role entry in the security index pertaining to the permissions here. When a plugin switches context from user context <-> plugin context an ephemeral (in-memory) role can be created with the privileges from this configuration. Likewise, I am wondering if a "plugin user" can also be ephemeral similar to how InjectSecurity works: https://github.com/opensearch-project/common-utils/blob/main/src/main/java/org/opensearch/commons/InjectSecurity.java#L157-L175

If its ephemeral then there would not be a need to delete a user on uninstallation or creating a user on installation.

Reply:
Hi @cwperks, that sounds like a great idea! I agree that plugin developers should be responsible for specifying the permissions associated with their applications. I think the same for extensions as well. 

Your example makes sense and seems to fit well with the existing system. I appreciate it avoids explicit role use. 

I will take a look at the link you shared but on the surface that sounds like a good idea.  

Currently, I am trying to find the best structuring of things since the node start process has become more complicated then expected based on how the existing pluginService class works and what information it needs to share with the IdentityPlugin. 

I like your idea though and will try to incorporate it. If you have any other suggestions let me know!